src/gui/src/staGui.cpp

```diff
@@ -249,127 +308,117 @@ bool TimingPathsModel::populatePaths(bool get_max,
+    auto* path_delay = path_end->pathDelay();
+    if (path_delay != nullptr) {
+      timing_path->setPathDelay(path_delay->delay());
+    } else {
+      timing_path->setPathDelay(0.0);
+    }
+    timing_path->setSlack(path_end->slack(sta));
+    timing_path->setPathArrivalTime(path_end->dataArrivalTime(sta));
+    timing_path->setPathRequiredTime(path_end->requiredTime(sta));
 
-    first_path = false;
+    bool clock_propagated = false;
+    if (start_clock_edge != nullptr) {
+      clock_propagated = start_clock_edge->clock()->isPropagated();
+    }
 
-    path->computeClkEndIndex();
+    const bool clock_expaneded = clock_propagated;
+
+    timing_path->populatePath(path,
+                              sta,
+                              dcalc_ap,
+                              clock_expaneded);
+    if (include_capture) {
+      timing_path->populateCapturePath(path_end->targetClkPath(),
+                                       sta,
+                                       dcalc_ap,
+                                       path_end->targetClkOffset(sta),
+                                       clock_expaneded);
+    }
+
+    timing_path->computeClkEndIndex();
+    timing_path->setSlackOnPathNodes();
 
-    timing_paths_.push_back(std::unique_ptr<TimingPath>(path));
+    paths.push_back(std::unique_ptr<TimingPath>(timing_path));
   }
-  QApplication::restoreOverrideCursor();
+
   delete path_ends;
-  return true;
 }
 
-/////////
-
-void TimingPath::populateNodeList(sta::Path* path, sta::dbSta* sta, sta::DcalcAnalysisPt* dcalc_ap, float offset, bool clock_expanded, bool first_path, TimingNodeList& list)
+void TimingPath::populateNodeList(sta::Path* path,
+                                  sta::dbSta* sta,
+                                  sta::DcalcAnalysisPt* dcalc_ap,
+                                  float offset,
+                                  bool clock_expanded,
+                                  TimingNodeList& list)
 {
   float arrival_prev_stage = 0;
   float arrival_cur_stage = 0;
 
   sta::PathExpanded expand(path, sta);
   for (size_t i = 0; i < expand.size(); i++) {
-    auto ref = expand.path(i);
-    auto pin = ref->vertex(sta)->pin();
-    auto slew = ref->slew(sta);
-    float cap = 0.0;
+    const auto* ref = expand.path(i);
+    const auto pin = ref->vertex(sta)->pin();
+    const bool pin_is_clock = sta->isClock(pin);
+    const auto slew = ref->slew(sta);
     const bool is_driver = sta->network()->isDriver(pin);
+    const auto is_rising = ref->transition(sta) == sta::RiseFall::rise();
+    const auto arrival = ref->arrival(sta);
+
+    float cap = 0.0;
     if (is_driver
         && !(!clock_expanded && (sta->network()->isCheckClk(pin) || !i))) {
       sta::Parasitic* parasitic = nullptr;
       sta::ArcDelayCalc* arc_delay_calc = sta->arcDelayCalc();
-      if (arc_delay_calc)
+      if (arc_delay_calc) {
```

Comment:
arc_delay_calc is not optional so the 'if' is pointless

Reply:
removed